core/src/main/java/org/jruby/ext/bigdecimal/RubyBigDecimal.java

```diff
@@ -1201,13 +1220,21 @@ public RubyBigDecimal op_pow(final ThreadContext context, IRubyObject exp) {
         return new RubyBigDecimal(runtime, pow);
     }
 
+    // mri bigdecimal_power_by_bigdecimal
+    private IRubyObject bigdecimal_power_by_bigdecimal(ThreadContext context, IRubyObject exp, int precision) {
+        RubyModule bigMath = context.getRuntime().getModule("BigMath");
+        RubyBigDecimal log_x = (RubyBigDecimal) bigMath.callMethod(context, "log", new IRubyObject[]{this, RubyFixnum.newFixnum(context.getRuntime(), precision + 1)});
+        RubyBigDecimal multipled = (RubyBigDecimal) log_x.mult2(context, exp, RubyFixnum.newFixnum(context.getRuntime(), precision + 1));
+        return bigMath.callMethod(context, "exp", new IRubyObject[]{multipled, RubyFixnum.newFixnum(context.getRuntime(), precision)});
+    }
+
     @Deprecated
     public IRubyObject op_pow19(IRubyObject exp) {
         return op_pow(getRuntime().getCurrentContext(), exp);
     }
 
     @Deprecated
-    public RubyBigDecimal op_pow19(ThreadContext context, IRubyObject exp) {
+    public IRubyObject op_pow19(ThreadContext context, IRubyObject exp) {
```

Comment:
we should either remove or not change signature for a deprecated method ...

Reply:
i select to remove this deprecated method.